sceptre/plan/actions.py

```diff
@@ -1000,3 +1000,115 @@ def diff(self, stack_differ):
         :rtype: sceptre.diffing.stack_differ.StackDiff
         """
         return stack_differ.diff(self)
+
+    @add_stack_hooks
+    def detect_stack_drift(self):
+        """
+        Detects stack drift for a running stack.
+
+        :returns: The stack drift.
+        :rtype: Tuple[str, Union[str, dict]]
+        """
+        response = self._detect_stack_drift()
+
+        detection_id = response["StackDriftDetectionId"]
+        status = self._wait_for_drift(detection_id)
+
+        if status == "DETECTION_COMPLETE":
+            response = self._describe_stack_resource_drifts()
```

Comment:
It looks like getting the drift detection status will return a StackDriftStatus, which will indicate if it's DRIFTED or IN_SYNC, or UNKNOWN. It would stand to reason that you wouldn't need to get individual resource drifts if the status is that it's IN_SYNC.

Reply:
I kind of like it the way it is though. A lot of info is returned, even when things are in sync. (See the screenshots I attached.) I think it would be annoying if you waited for 20 seconds and then the info returned from AWS is discarded because things were in sync.

Reply:
I can respect that. `validate` also provides a bunch of output, even when things are valid.